WalletLibrary/WalletLibrary/Utilities/Dictionary+ValueExtraction.swift

```diff
@@ -16,12 +16,22 @@ extension Dictionary
             if let key = key as? String,
                let value = value as? String
             {
-                let part = "\(key)=\(value)".addingPercentEncoding(withAllowedCharacters: .urlQueryAllowed) ?? ""
+                let part = "\(urlEncode(key))=\(urlEncode(value))"
                 parts.append(part)
             }
         }
         return parts.joined(separator: "&")
     }
+    
+    private func urlEncode(_ string: String) -> String 
+    {
+        // Define the allowed characters according to RFC 3986
+        var allowedCharacterSet = CharacterSet.alphanumerics
+        allowedCharacterSet.insert(charactersIn: "-._*")
```

Comment:
Given that this is the token endpoint URL and that the name of this method is URL, we should probably stick to the [URL RFC](https://www.[ietf.org/rfc/rfc1738.txt](https://www.ietf.org/rfc/rfc1738.txt)) rather than the URI RFC and be a bit more lenient:

"Thus, only alphanumerics, the special characters "$-_.+!*'(),", and reserved characters used for their reserved purposes may be used unencoded within a URL."

```suggestion
        allowedCharacterSet.insert(charactersIn: "$-_.+!*'(),")
```

Reply:
good idea.